aptos-move/mvhashmap/src/versioned_aggregators.rs

```diff
@@ -4,53 +4,60 @@
 use crate::types::{AtomicTxnIndex, MVAggregatorsError, TxnIndex};
 use aptos_aggregator::{
     aggregator_change_set::{AggregatorApplyChange, ApplyBase},
-    types::{AggregatorID, AggregatorValue},
+    types::AggregatorValue,
 };
 use claims::{assert_matches, assert_none};
 use crossbeam::utils::CachePadded;
 use dashmap::DashMap;
 use std::{
     collections::btree_map::{BTreeMap, Entry},
+    fmt::Debug,
+    hash::Hash,
     iter::DoubleEndedIterator,
     sync::atomic::Ordering,
 };
 
+pub enum CommitError {
+    CodeInvariantError(String),
+    ReExecutionNeeded(String),
+}
+
 // When an AggregatorEntry (see below) is transformed to an Estimate, internally we store
 // a potential bypass (based on the previously stored entry), which may allow a read
 // operation to not wait for the corresponding dependency.
 #[derive(Clone, Debug, PartialEq)]
-enum EstimatedEntry {
+enum EstimatedEntry<K: Clone> {
     NoBypass,
     // If applicable, can bypass the Estimate by considering a apply change instead.
-    Bypass(AggregatorApplyChange),
+    Bypass(AggregatorApplyChange<K>),
 }
 
 // There is no explicit deletion as it will be impossible to resolve the ID of a deleted
 // aggregator (as the ID will not be contained in a resource anymore). The write on the
 // previously holding resource and Block-STM read validation ensures correctness.
 #[derive(Debug)]
-enum AggregatorEntry {
+enum AggregatorEntry<K: Clone> {
```

Comment:
AggregatorEntry and AggregatorValue sounds alike. Should we rename AggregatorEntry to AggregatorStorageEntry or AggregatorMVHashmapEntry?

Reply:
updated to VersionEntry (as it represents a single version value)
(in contrast to VersionedValue to represent multiple versions)

this is local to this file, so we can change it later if people have other preferences